### PR TITLE
Entity type4: Remove MiscEntityType and add FountainType

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Final few bits of reorg of the fundamental setup of entities. Removal of SubType as we can just use Type but for jumping fountains we needed a way to distinguish between snow and water so had to add a field for that.

### CODE REVIEW

File: src/openrct2/world/Balloon.cpp [hunk 1]
```diff
@@ -81,11 +81,7 @@ void Balloon::Pop()
...
 void create_balloon(const CoordsXYZ& balloonPos, int32_t colour, bool isPopped)
 {
-    auto* sprite = CreateEntity<Balloon>();
-    if (sprite == nullptr)
-        return;
-    sprite->SubType = MiscEntityType::Balloon;
-    auto balloon = sprite->As<Balloon>();
+    auto* balloon = CreateEntity<Balloon>();
     if (balloon == nullptr)
         return; // can never happen
```
Comment (Gymnasiast) [inline src/openrct2/world/Balloon.cpp:86]: I think this comment is now no longer correct, right?

File: src/openrct2/paint/sprite/Paint.Misc.cpp [hunk 2]
```diff
@@ -113,7 +113,7 @@ template<> void PaintEntity(paint_session* session, const JumpingFountain* jumpi
         isAntiClockwise = !isAntiClockwise;
     }
 
-    uint32_t baseImageId = (jumpingFountain->SubType == MiscEntityType::JumpingFountainSnow) ? 23037 : 22973;
+    uint32_t baseImageId = (jumpingFountain->FountainType == JumpingFountainType::Snow) ? 23037 : 22973;
```
Comment (Gymnasiast) [inline src/openrct2/paint/sprite/Paint.Misc.cpp:116]: Would like to have constants for these images.

**Comment by duncanspumpkin** [comment]
`JUMPING_FOUNTAIN_TYPE_SNOW` no longer exists it was the basis for the new enum class.

**Comment by duncanspumpkin** [comment]
Requires another network version bump prior to merge

**Review by Gymnasiast — COMMENTED** [review]
Code looks good, would like to do a little test.

**Review by Gymnasiast — APPROVED** [review]
Appears to work correctly.

**Comment by Gymnasiast** [comment]
Mind the milestone.

**Pull request merged**